File: BTrees/examples/b_tree_example.cpp
#include "../source/b_tree.h"
#include <iostream>

int main() {
    // Initialize a B-Tree with a minimum degree of 3
    BTree bTree(3);

    // Insert keys into the B-Tree
    std::cout << "Inserting keys: 10, 20, 5, 6, 12, 30, 7, 17" << std::endl;
    bTree.insert(10);
    bTree.insert(20);
    bTree.insert(5);
    bTree.insert(6);
    bTree.insert(12);
    bTree.insert(30);
    bTree.insert(7);
    bTree.insert(17);

    // Print the B-Tree structure after insertion
    std::cout << "\nB-Tree structure after insertions:" << std::endl;
    bTree.printTree();

    // Search for a specific key in the B-Tree
    int searchKey = 12;
    std::cout << "\nSearching for key " << searchKey << "..." << std::endl;
    if (bTree.search(searchKey)) {
        std::cout << "Key " << searchKey << " found in the B-Tree." << std::endl;
    } else {
        std::cout << "Key " << searchKey << " not found in the B-Tree." << std::endl;
    }

    // Delete a key from the B-Tree
    int deleteKey = 6;
    std::cout << "\nDeleting key " << deleteKey << " from the B-Tree..." << std::endl;
    bTree.remove(deleteKey);

    // Print the B-Tree structure after deletion
    std::cout << "\nB-Tree structure after deletion:" << std::endl;
    bTree.printTree();

    return 0;
}

File: BTrees/examples/b_tree_example.js
const BTree = require('../source/b_tree'); // Adjust the path if needed

// Initialize a B-Tree with a minimum degree of 3
const bTree = new BTree(3);

console.log("Inserting keys into the B-Tree...");
[10, 20, 5, 6, 12, 30, 7, 17].forEach((key) => {
    bTree.insert(key);
    console.log(`Inserted key: ${key}`);
});

// Print the B-Tree structure after insertions
console.log("\nB-Tree structure after insertions:");
bTree.printTree();

// Search for keys in the B-Tree
console.log("\nSearching for keys in the B-Tree...");
[6, 15, 30].forEach((key) => {
    const result = bTree.search(key);
    if (result) {
        console.log(`Key ${key} found in node: ${result.keys}`);
    } else {
        console.log(`Key ${key} not found in the B-Tree.`);
    }
});

// Remove keys from the B-Tree
console.log("\nDeleting keys from the B-Tree...");
[6, 20, 30].forEach((key) => {
    bTree.remove(key);
    console.log(`Deleted key: ${key}`);
    console.log("Updated B-Tree structure:");
    bTree.printTree();
});
